lib/ansible/playbook/task.py

```diff
@@ -106,6 +107,90 @@ def __init__(self, block=None, role=None, task_include=None):
 
         super(Task, self).__init__()
 
+    def _validate_module_defaults(self, attribute, name, value):
+        # load, resolve, and cache any action groups the task may use
+        if value is None:
+            return
+
+        for defaults_dict in value:
+            for defaults_entry, defaults in defaults_dict.items():
+                if not defaults_entry.startswith('group/'):
+                    continue
+
+                group_name = defaults_entry.split('group/')[-1]
+                collection_name = '.'.join(group_name.split('.')[0:2])
+                self._resolve_group(group_name, collection_name)
+
+        # _load_module_defaults in the base class ensures value is a list of dictionaries,
+        # the keys of which are fully qualified action and group names
+        setattr(self, name, value)
+
+    def _resolve_group(self, group, collection_name):
```

Comment:
This can seemingly be moved to `Base` (along with moving the calls to `_load_module_defaults`)- the only tricky bit is getting at the play cache when you're a play. This seemed to work in my messings around:
`play = getattr(self, '_play', self)`

Reply:
Task/TaskIncludes need to get the play from the parent. Maybe I'll add a play property to the Base class? Or, my initial approach was to add cache properties for the Task and Block classes.

```
    @property
    def play(self): 
        if hasattr(self, '_play'):
            play = self._play
        elif hasattr(self, '_parent') and hasattr(self._parent, '_play'):
            play = self._parent._play
        else:
            play = self

        if play.__class__.__name__  != 'Play':
            # Should not happen
            # raise NotImplementedError("Failed to find 'play' for module_defaults")
            return None   

        return play  
```

Reply:
At a glance, the property LGTM- just have to make sure it doesn't get used in the future by anything internal to this too early (eg if a future `Base` validator method or something called it before the relevant FA values used by the property impl have been set), but I think those are set pretty early anyway, so probably pretty safe. You might want to see if, eg @sivel or @jimi-c disagree on that though.

Reply:
I don't see a problem with the proposed property.